framework/contracts/account/tests/xion.rs

```diff
@@ -123,3 +142,116 @@ fn test_init_sign_arb() {
     )
     .unwrap();
 }
+
+/// Initialize the account with the test owner as the owner
+pub(crate) fn mock_init(
+    deps: &mut OwnedDeps<MockStorage, MockApi, MockQuerier, Empty>,
+) -> AccountResult {
+    let abstr = AbstractMockAddrs::new(deps.api);
+
+    let info = message_info(&abstr.owner, &[]);
+    let env = mock_env_validated(deps.api);
+
+    abstract_account::contract::instantiate(
+        deps.as_mut(),
+        env,
+        info,
+        account::InstantiateMsg {
+            code_id: 1,
+            account_id: Some(AccountId::new(1, AccountTrace::Local).unwrap()),
+            owner: GovernanceDetails::Monarchy {
+                monarch: abstr.owner.to_string(),
+            },
+            namespace: None,
+            name: Some("test".to_string()),
+            description: None,
+            link: None,
+            install_modules: vec![],
+            authenticator: None,
+        },
+    )
+}
+
+const OWNERSHIP: Item<Ownership<Addr>> = Item::new(OWNERSHIP_STORAGE_KEY);
+
+#[test]
+fn xion_account_auth_itself() -> anyhow::Result<()> {
+    let mut deps = mock_dependencies();
+    deps.querier = abstract_mock_querier(deps.api);
+    mock_init(&mut deps)?;
+
+    let env = mock_env_validated(deps.api);
+    // We set the contract as owner.
+    // We can't make it through execute msgs, because of XION signatures are too messy to reproduce in tests
+    let ownership = Ownership {
+        owner: GovernanceDetails::AbstractAccount {
+            address: env.contract.address.clone(),
+        }
+        .verify(deps.as_ref())?,
+        pending_owner: None,
+        pending_expiry: None,
+    };
+    OWNERSHIP.save(deps.as_mut().storage, &ownership)?;
+
+    let whitelisted = deps.api.addr_make("whitelisted");
+    let not_whitelisted_yet = deps.api.addr_make("not_whitelisted");
+
+    // We whitelist a module
+    AUTH_ADMIN.save(deps.as_mut().storage, &true)?;
```

Comment:
Do we need to clear this admin flag before executing the not_whitelisted update messages?

Reply:
The Admin flag is cleared at the end of the execute function